lib/graphql/query/serial_execution/field_resolution.rb

```diff
@@ -47,21 +53,35 @@ def get_finished_value(raw_value)
             end
           end
 
-          begin
-            GraphQL::Query::SerialExecution::ValueResolution.resolve(
-              parent_type,
-              field,
-              field.type,
-              raw_value,
-              @irep_nodes,
-              @field_ctx,
-            )
-          rescue GraphQL::InvalidNullError => err
-            if field.type.kind.non_null?
-              raise(err)
-            else
-              err.parent_error? || @query.context.errors.push(err)
-              nil
+          box_method = @query.boxed_value_method(raw_value)
+          if box_method
+            GraphQL::Execution::Boxed.new { raw_value.public_send(box_method) }.then { |val|
+              GraphQL::Query::SerialExecution::ValueResolution.resolve(
+                parent_type,
+                field,
+                field.type,
+                val,
+                @irep_nodes,
+                @field_ctx,
+              )
+            }
+          else
+            begin
+              GraphQL::Query::SerialExecution::ValueResolution.resolve(
+                parent_type,
+                field,
+                field.type,
+                raw_value,
+                @irep_nodes,
+                @field_ctx,
+              )
+            rescue GraphQL::InvalidNullError => err
```

Comment:
> Each spot that the gem captured errors "for you" has come back to bite me.

I didn't mean for internal errors, I meant in cases like this where there is already exception handling logic.  For instance, if a non-null field raises a GraphQL::ExecutionError, then that exception wouldn't be caught unless done by the registered unboxing method, and even if that method catches and returns the error it will result in a `GraphQL::InvalidNullError` that won't get caught by parent nullable fields.

However, maybe this could be simplified by doing this GraphQL::InvalidNullError error propagation by returning the error rather than raising the exception.